website/docs/help-and-support/troubleshooting-guide/action-errors/datasource-errors.md

```diff
@@ -66,14 +66,32 @@ At least one of the mandatory fields in the plugin's datasource creation form is
 
 This message indicates that one of the mandatory fields, for example, `Access Key`, has been left empty in the [datasource creation form](../../../core-concepts/connecting-to-data-sources/connecting-to-databases.md).
 
-This error can be fixed by filling the mentioned mandatory field(s) in the [datasource creation form](../../../core-concepts/connecting-to-data-sources/connecting-to-databases.md).
+This error can be fixed by filling the mentioned mandatory fields in the [datasource creation form](../../../core-concepts/connecting-to-data-sources/connecting-to-databases.md).
 
-### Cannot Delete Datasource Error
+### Can't delete datasource error
 
 ```
 Cannot delete datasource since it has 1 action(s) using it.
 ```
 
-This message indicates that the [datasource](../../../reference/datasources/) attempting to be deleted has some query action configured on it.
+This message indicates that the [datasource](/reference/datasources/) attempting to be deleted has some query action configured on it.
 
-This error can be fixed by deleting any queries dependent on this [datasource](../../../reference/datasources/) before attempting to delete the [datasource](../../../reference/datasources/).
+This error can be fixed by deleting any queries dependent on this [datasource](/reference/datasources/) before attempting to delete the [datasource](../../../reference/datasources/).
+
+### Error connecting to local db or API
+
+If you are trying to connect to a local MySQL database from Appsmith and see an error message like:
+
+ ```
+ io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /172.17.0.1:3306
+ ```
+
+When running Appsmith inside a Docker container, it may have its own network namespace and won't be able to access services running on the host machine using the ```localhost``` or ```127.0.0.1``` addresses. This is because these addresses are reserved for use within the network namespace of the container, and are not accessible from the host machine.
+
+Instead, you can use the hostname ```host.docker.internal``` on **Windows** and **macOS**, and ```172.17.0.1``` on **Linux**, to access services running on the host machine from within the container. This will allow the container to access the MySQL server running on the host.
+
+It's important to note that binding the MySQL server to ```0.0.0.0``` will allow connections from any host, including other devices on the same network as the host. This may or may not be desirable, depending on your security requirements.
```

Comment:
```suggestion
In particular, if you are connecting to a MySQL server, make sure that it's configured to bind to `0.0.0.0`. This will allow connections from any host, including other devices on the same network. This may or may not be desirable, depending on your security requirements.
```